modules/openapi/build.gradle.kts

```diff
@@ -0,0 +1,65 @@
+plugins {
+    kotlin("jvm") version "2.0.0"
+    id("org.openapi.generator") version "6.2.1"
+}
+
+group = "com.sphereon.oid.fed"
+version = "0.1.0-SNAPSHOT"
+
+project.extra.set("openApiPackage", "com.sphereon.oid.fed.openapi")
+
+val profiles = project.properties["profiles"]?.toString()?.split(",") ?: emptyList()
+val isModelsOnlyProfile = profiles.contains("models-only")
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation(kotlin("test"))
+}
+
+openApiGenerate {
+    val openApiPackage: String by project
+    generatorName.set("kotlin")
+    packageName.set("com.sphereon.oid.fed.openapi")
+    apiPackage.set("$openApiPackage.api")
+    modelPackage.set("$openApiPackage.models")
+    inputSpec.set("$projectDir/src/main/kotlin/com/sphereon/oid/fed/openapi/openapi.yaml")
+    library.set("jvm-okhttp4")
```

Comment:
Why are we using a jvm target library for this?
Please use library.set("multiplatform")
 example:
 ```
 openApiGenerate {
    generatorName.set("kotlin")
    library.set("multiplatform")
    inputSpec.set("path/to/your/openapi/spec.yaml")
    outputDir.set("$buildDir/generated")
    additionalProperties.set([
        "serializationLibrary": "kotlinx_serialization",
        "dateLibrary": "string"
    ])
}
```

Reply:
Are you sure about additional properties and outputDir? They are not present in the kotllin documentation: https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/kotlin.md. Besides the array passed to additionalProperties generates a compilation error because it's not kotlin code.

Reply:
This was just a copy-paste example. We need to set outputDir wherever we need to go if the default is not what we want.
For .kts it should look something like this:
```
plugins {
    kotlin("multiplatform")
    id("org.openapi.generator") version "6.2.1" // Use the latest version available
}


openApiGenerate {
   generatorName.set("kotlin")
   library.set("multiplatform")
   inputSpec.set("path/to/your/openapi/spec.yaml")
   outputDir.set("$buildDir/generated")
   additionalProperties.set([
       "serializationLibrary": "kotlinx_serialization",
       "dateLibrary": "string"
   ])
}
```

Reply:
BTW, adding output dir breaks the build. The right place to set additional properties is there :point_down:. However kotlin_serialization and string as values break the build. Please don't make me waste time with chatGPT inaccurate answers.

Reply:
Just by setting library to "multiplatform" made it use kotlinx serialization and ktor client